packages/api/src/controllers/insightHelpers.js

```diff
@@ -236,105 +238,100 @@ export const getBiodiversityAPI = async (pointData, countData) => {
     data: [],
   };
 
-  const dictionary = {
-    Aves: 'Birds',
-    Insecta: 'Insects',
-    Plantae: 'Plants',
-    Amphibia: 'Amphibians',
-    Mammalia: 'Mammals',
+  const filterDictionary = {
+    Birds: { classKey: 212 },
+    Insects: { classKey: 216 },
+    Plants: { kingdomKey: 6 },
+    Amphibians: { classKey: 131 },
+    Mammals: { classKey: 359 },
   };
 
   const speciesCount = {
     Birds: 0,
     Insects: 0,
     Plants: 0,
     Amphibians: 0,
-    CropVarieties: 0,
+    CropVarieties: parseInt(countData),
     Mammals: 0,
   };
-  speciesCount['CropVarieties'] = parseInt(countData);
+
+  const parsedSpecies = [];
+
+  const processBiodiversityResults = (results, label) => {
+    results.map((result) => {
```

Comment:
Nothing is returned from this function, so I think `forEach` should be used instead.